test/e2e/provisioning/helper.go

```diff
@@ -97,56 +99,46 @@ func doesSenarioSelectorMatch(selector *scenarioSelector, testCase scenario) boo
+	buildNum := os.Getenv("CIRCLE_BUILD_NUM")
+	if buildNum == "" {
+		buildNum = "local"
+	}
+
 	for _, testCase := range scenarios {
 		if excludeSelector != nil && doesSenarioSelectorMatch(excludeSelector, testCase) {
 			continue
 		}
 
-		testCase := testCase // capture range variable
-		t.Run(testCase.name, func(t *testing.T) {
-			t.Parallel()
-
-			nameSufix := strings.Replace(testCase.name, " ", "-", -1)
-			nameSufix = strings.ToLower(nameSufix)
-			nameSufix = fmt.Sprintf("%s-%s", cloudProvider, nameSufix)
-			machineName := fmt.Sprintf("machine-%s", nameSufix)
-			nodeName := fmt.Sprintf("node-%s", nameSufix)
-			params := testParams
-			params = fmt.Sprintf("%s,<< MACHINE_NAME >>=%s,<< NODE_NAME >>=%s-%s", params, machineName, nodeName, os.Getenv("CIRCLE_BUILD_NUM"))
-			params = fmt.Sprintf("%s,<< OS_NAME >>=%s,<< CONTAINER_RUNTIME >>=%s,<< CONTAINER_RUNTIME_VERSION >>=%s", params, testCase.osName, testCase.containerRuntime, testCase.containerRuntimeVersion)
-
-			err := runVerifyTool(manifestPath, params)
-			if err != nil {
-				t.Errorf("verify tool failed due to %v", err)
-			}
+		st.Run(testCase.name, func(it *testing.T) {
+			testScenario(it, testCase, buildNum, cloudProvider, testParams, manifestPath)
 		})
 	}
 
 }
 
-func runVerifyTool(manifestPath string, params string) error {
+func testScenario(t *testing.T, testCase scenario, buildNum, cloudProvider string, testParams []string, manifestPath string) {
+	t.Parallel()
+
+	nameSufix := strings.Replace(testCase.name, " ", "-", -1)
+	nameSufix = strings.ToLower(nameSufix)
+	nameSufix = fmt.Sprintf("%s-%s-%s", cloudProvider, nameSufix, buildNum)
+	nodeName := fmt.Sprintf("node-%s", nameSufix)
+	machineName := fmt.Sprintf("machine-%s", nameSufix)
+	scenarioParams := append([]string(nil), testParams...)
+	scenarioParams = append(scenarioParams, fmt.Sprintf("<< MACHINE_NAME >>=%s", machineName))
+	scenarioParams = append(scenarioParams, fmt.Sprintf("<< NODE_NAME >>=%s", nodeName))
+	scenarioParams = append(scenarioParams, fmt.Sprintf("<< OS_NAME >>=%s", testCase.osName))
+	scenarioParams = append(scenarioParams, fmt.Sprintf("<< CONTAINER_RUNTIME >>=%s", testCase.containerRuntime))
+	scenarioParams = append(scenarioParams, fmt.Sprintf("<< CONTAINER_RUNTIME_VERSION >>=%s", testCase.containerRuntimeVersion))
+
 	gopath := os.Getenv("GOPATH")
 	projectDir := filepath.Join(gopath, "src/github.com/kubermatic/machine-controller")
 
 	kubeConfig := filepath.Join(projectDir, ".kubeconfig")
-	args := []string{
-		"-input",
-		manifestPath,
-		"-parameters",
-		params,
-		"-kubeconfig",
-		kubeConfig,
-		"-machineReadyTimeout",
-		"40m",
-		"-logtostderr",
-		"true",
-	}
 
-	cmd := exec.Command(filepath.Join(projectDir, "test/tools/verify/verify"), args...)
-	out, err := cmd.CombinedOutput()
+	err := verifyhelper.Verify(kubeConfig, manifestPath, scenarioParams, false, 60*time.Hour)
```

Comment:
setting the timeout to 60 hours seems to be too much :)

Reply:
tests will timeout after `20` minutes

Reply:
Yeah the idea here was to have a timeout that is not going to kick in because the overall timeout is supposed to be used